rasa/core/actions/action.py

```diff
@@ -591,7 +591,103 @@ def action_response_format_spec() -> Dict[Text, Any]:
                     "type": "array",
                     "items": {
                         "type": "object",
-                        "properties": {"event": {"type": "string"}},
+                        "properties": {"event": {"type": "string"}, "timestamp": {}},
+                        "oneOf": [
+                            {
+                                "properties": {
+                                    "event": {"const": "user"},
+                                    "text": {"type": "string"},
+                                    "input_channel": {},
+                                    "message_id": {"type": "string"},
+                                    "parse_data": {
+                                        "type": "object",
+                                        "properties": {
+                                            "text": {"type": "string"},
+                                            "intent": {
+                                                "type": "object",
+                                                "properties": {
+                                                    "name": {"type": "string"},
+                                                    "confidence": {"type": "number"},
+                                                },
+                                            },
+                                            "entities": {
+                                                "type": "array",
+                                                "items": {
+                                                    "type": "object",
+                                                    "properties": {
+                                                        "start": {"type": "integer"},
+                                                        "end": {"type": "integer"},
+                                                        "name": {"type": "string"},
+                                                        "confidence": {
+                                                            "type": "number"
+                                                        },
+                                                        "extractor": {"type": "string"},
+                                                        "value": {"type": "object"},
+                                                    },
+                                                },
+                                            },
+                                        },
+                                    },
+                                }
+                            },
+                            {
+                                "properties": {
+                                    "event": {"const": "action"},
+                                    "policy": {},
```

Comment:
that's optional text, isn't it"

Reply:
For `policy`? Yes, it's actually type `Any`, I couldn't find the equivalent of Optional in json schema, not sure if it's supported.

Reply:
Pretty sure it can only be a string or null. Where did you find `Any`? 

Apparently you can do `Optional` as described [here](https://stackoverflow.com/a/47592496/3429596)

Reply:
Ah nice find! The `Any` type solution I sourced from first answer [here.](https://stackoverflow.com/questions/16826128/why-is-this-json-schema-invalid-using-any-type/16826176) ... I should have scrolled below and I could have found the multiple types solution you shared 🤦‍♀️